File: w10/d02/morning/find_substring_2.rb
def find_substring(string, substring)
  0.upto(string.length - substring.length) do |i|
    return i if substring == string[i..i+substring.length-1]
  end
  false
end

def find_substring2(string, substring)
  string =~ (/substring/) || false
end

def find_substring3(string, substring)
  string.match(substring) ? string.match(substring).begin(0) : false
end

File: w10/d02/morning/find_substring_2.js
function findSubstring(string, substring) {
  result = string.search(substring)
  if (result === -1) {
    return false
  } else {
    return result
  }
}

function findSubstring(string, substring) {
  for (var i = 0; i < string.length - substring.length; i++) {
    if (substring === string.substr(i, i+substring.length)) {
      return i
    }
  }
  return false
}
